mitmproxy/addons/modifyheaders.py

```diff
@@ -7,92 +7,68 @@
-    subject: bytes
-    replacement: str
-
-    def read_replacement(self) -> bytes:
-        """
-        Process the replacement str. This usually just involves converting it to bytes.
-        However, if it starts with `@`, we interpret the rest as a file path to read from.
-
-        Raises:
-            - IOError if the file cannot be read.
-        """
-        if self.replacement.startswith("@"):
-            return Path(self.replacement[1:]).expanduser().read_bytes()
-        else:
-            # We could cache this at some point, but unlikely to be a problem.
-            return strutils.escaped_str_to_bytes(self.replacement)
-
-
 def _match_all(flow) -> bool:
     return True
 
 
-def parse_modify_spec(option, subject_is_regex: bool, replacement_is_path: bool) -> ModifySpec:
+def parse_spec(option: str) -> typing.Tuple[flowfilter.TFilter, str, str]:
     """
-        The form for the modify_*, map_remote, and map_local options is as follows:
-
-            * modify_body:    [/flow-filter]/body-regex/[@]replacement
-            * modify_headers: [/flow-filter]/header-name/[@]header-value
-            * map_local:      [:flow-filter]:url-regex:path
-            * map_remote:     [:flow-filter]:url-regex:[@]replacement
-
-        The @ allows to provide a file path that is used to read the respective option.
-        The addons ModifyHeaders, ModifyBody, MapRemote, and MapLocal use ModifySpec
-        to represent a single rule.
-
-        The first character specifies the separator. Example:
-
-            :~q:foo:bar
-
-        If only two clauses are specified, the flow filter is set to
-        match universally (i.e. ".*"). Example:
-
-            /foo/bar
+    Parse strings in the following format:
 
-        Clauses are parsed from left to right. Extra separators are taken to be
-        part of the final clause. For instance, the last parameter (header-value,
-        replace, or path) below is "foo/bar/":
+        [/flow-filter]/subject/replacement
 
-            /one/two/foo/bar/
     """
     sep, rem = option[0], option[1:]
     parts = rem.split(sep, 2)
     if len(parts) == 2:
-        flow_filter = _match_all
         subject, replacement = parts
+        return _match_all, subject, replacement
     elif len(parts) == 3:
-        flow_filter_pattern, subject, replacement = parts
-        flow_filter = flowfilter.parse(flow_filter_pattern)  # type: ignore
+        patt, subject, replacement = parts
+        flow_filter = flowfilter.parse(patt)
         if not flow_filter:
-            raise ValueError(f"Invalid filter pattern: {flow_filter_pattern}")
+            raise ValueError(f"Invalid filter pattern: {patt}")
+        return flow_filter, subject, replacement
     else:
         raise ValueError("Invalid number of parameters (2 or 3 are expected)")
 
-    subject = strutils.escaped_str_to_bytes(subject)
+
+class ModifySpec(typing.NamedTuple):
+    matches: flowfilter.TFilter
+    subject: bytes
+    replacement_str: str
```

Comment:
Minor: You changed this back from `replacement` to `replacement_str`.
Why include the type only for this one, and not for e.g. the subject? In `MapRemoteSpec` we use `replacement`.

Reply:
Ideally I would call it `_replacement` but NamedTuple doesn't like that. The idea is that you don't want to access `replacement_str` directly, but always use `read_replacement()`.